CMakeLists.txt

```diff
@@ -235,12 +235,16 @@ else()
     message(FATAL_ERROR "Qt version ${Launcher_QT_VERSION_MAJOR} is not supported")
 endif()
 
-include(ECMQueryQt)
-ecm_query_qt(QT_PLUGINS_DIR  QT_INSTALL_PLUGINS)
-ecm_query_qt(QT_LIBS_DIR     QT_INSTALL_LIBS)
-ecm_query_qt(QT_LIBEXECS_DIR QT_INSTALL_LIBEXECS)
-ecm_query_qt(QT_DATA_DIR     QT_HOST_DATA)
-set(QT_MKSPECS_DIR ${QT_DATA_DIR}/mkspecs)
+if(Launcher_QT_VERSION_MAJOR EQUAL 5)
+    include(ECMQueryQt)
+    ecm_query_qt(QT_PLUGINS_DIR  QT_INSTALL_PLUGINS)
+    ecm_query_qt(QT_LIBS_DIR     QT_INSTALL_LIBS)
+    ecm_query_qt(QT_LIBEXECS_DIR QT_INSTALL_LIBEXECS)
+else()
+    set(QT_PLUGINS_DIR  ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_PLUGINS})
+    set(QT_LIBS_DIR     ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_LIBS})
+    set(QT_LIBEXECS_DIR ${QT${QT_VERSION_MAJOR}_INSTALL_PREFIX}/${QT${QT_VERSION_MAJOR}_INSTALL_LIBEXECS})
```

Comment:
What's up with this? ECMQueryQt supports Qt 6 paths.
I think this makes the code less readable

Reply:
it breaks on windows on arm cross compilation

Reply:
The issue occurs with cross compiling for arm, Qt uses both an install for the Host architecture (x64) and for the target/install architecture (Arm64)

ECMQueryQt however seems to pick qtpaths in the host Qt directory instead of the one in the Install Qt directory, this results in you getting paths for applicable host Qt, resulting in x64 Qt dlls in the arm64 PrismLauncher bundles. (This issue occurs with both the official Qt installs and the aqtinstalls

Additionally, the msvc arm64 qtpaths is actually a batch script to launch qtpaths from host Qt with a config for the correct paths, which aqtinstall doesn't setup correctly.

Reply:
In this case you could remove `cmake/ECMQueryQt.cmake`, as it was only included in-tree for Qt 6 support

Reply:
done

Reply:
seems to fail without it :/

Reply:
seems like ubuntu 20.04 uses ecm 5.92 but that was added in 5.93, gonna revert